raycaster.h

```diff
@@ -35,7 +36,10 @@ class RayCaster
                        uint16_t *textureY,
                        uint16_t *textureStep) = 0;
 
-    RayCaster(){};
+    RayCaster(Game *g) : game(g){};
```

Comment:
Game object would be decoupled from `RayCaster` base class for the following considerations:
1. The derived classes such as `RayCasterFixed` and `RayCasterFloat` maintain the internal states/scenes without the exposure to Game object.
2. The Game object is user-oriented, applying the player actions without the understanding for class RayCaster.

Reply:
+ Do you mean that the Game object shouldn't be passed as a member of RayCaster? If so, then I can move `IsWall` to `raycaster_data.h`(or should I put it in a .cpp?), where both `Game` and `RayCaster` have access to.

Reply:
Exactly, always decouple Game object from RayCaster and its derived objects. File `raycaster_data.h` is intended to be generated and preserved, so please don't touch that.

Reply:
@jserv :
I'm thinking about wrapping `IsWall` together with the map data in a class `Map`. The `Game` object then will have a `Map` object as its member, which can be passed to `RayCaster` by the `Renderer`. This way the `RayCaster` doesn't need a direct handle to the `Game` object. This also allows possible feature like multiple maps and switching map is as easy as changing it in `Game`. Does that sounds good to you?

Reply:
Send another pull request to address the refinement of map oriented object(s).